lib/rules/no-async-describe.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+/* eslint "complexity": [ "error", 5 ] */
+
+/**
+ * @fileoverview Disallow async functions as arguments to describe
+ */
+
+const astUtils = require('../util/ast');
+const { additionalSuiteNames } = require('../util/settings');
+
+module.exports = function (context) {
+    const sourceCode = context.getSourceCode();
+
+    function isFunction(node) {
+        return (
+            node.type === 'FunctionExpression' ||
+            node.type === 'FunctionDeclaration' ||
+            node.type === 'ArrowFunctionExpression'
+        );
+    }
+
+    function containsDirectAwait(node) {
```

Comment:
I don't know if this could be implemented with a funky ramda function that looks for a nested object with a `type` of `AwaitExpression` not contained within one with a `type` of `FunctionExpression` etc. -- but I have to admit that I'm not really familiar with that.

Reply:
Don’t worry, it’s not a requirement to write funky ramda expressions 😉.